lib/image_processing/vips.rb

```diff
@@ -0,0 +1,231 @@
+require "vips"
+require_relative "vips/color"
+require_relative "vips/gravity"
+require "tempfile"
+
+if Gem::Version.new(Vips::VERSION) < Gem::Version.new("2.0.0")
+  raise "image_processing requires ruby-vips version >= 2.0.0"
+end
+
+module ImageProcessing
+  module Vips
+    module_function
+
+    # Changes the image encoding format to the given format
+    # Libvips convert the file automatically based on the name
+    # since we want the user to have a destructive and a non destructive method
+    # using the destructive method will cause the original file to be deleted
+    #
+    # @param [File, Tempfile] file     the image to convert
+    # @param [String] format           the format to convert to
+    # @param [File] destination        the end destination where the image will be safe
+    # @return [File, Tempfile]
+    def convert(file, format, destination: nil, &block)
+      vips_image = ::Vips::Image.new_from_file file.path
+      destination_image = destination || _tempfile(".#{format}")
+      vips_image.write_to_file(destination_image.path)
+      destination_image
+    end
+
+    def convert!(file, format, &block)
+      destination_path = Pathname(file.path).sub_ext(".#{format}").to_s
+      convert(image, format, destination: File.new(destination_path), &block).tap do
```

Comment:
I think having `destination` be a `File` object won't work, as at the moment of conversion the destination path will not exist in `#convert!` or when users choose to specify a custom path. We should probably use a String path for `:destination` instead.

Reply:
So you think is better the destination to be `String path` just in `convert` and `convert!` or for all the methods.

Also, this part confuses me.
>  as at the moment of conversion the destination path will not exist in #convert!

Reply:
> Also, this part confuses me.

>> as at the moment of conversion the destination path will not exist in #convert!

What I wanted to say is that `#convert!` is the only bang processing method where the destination path isn't the same as the source. Unlike other bang processing methods, it doesn't modify the file in place, because it writes the result into a new extension.

If you would write a test for `#convert!` as it is implemented now, it would fail on `File.new(destination_path)`, saying that the file doesn't exist.

The reason why I mentioned `#convert!` is because it shows that using a `File` object for destination probably isn't a good design choice after all, because it would complicate `#convert!` implementation.

Reply:
You are totally right! Lets use `Strings`